File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let extent = brown + yellow;
    
    for (let height = 3; height <= Math.sqrt(extent); height++) {
        if (extent % height === 0) {
            let width = extent / height;
            if ((width - 2) * (height - 2) === yellow) {
                return [width, height];
            }
        }
    }
    return [];
}

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(int brown, int yellow) {
    int extent = brown + yellow;

    for (int height = 3; height <= (int)sqrt(extent); ++height) {
        if (extent % height == 0) {
            int width = extent / height;

            if ((width - 2) * (height - 2) == yellow) {
                return {width, height};
            }
        }
    }
    return {};
}
